Makefile

```diff
@@ -136,7 +160,7 @@ pre-commit: fmt lint
 
 # Downloads envoy, based on the SHA defined in the base pilot Dockerfile
 # Will also check vendor, based on Gopkg.lock
-init:
+init: check-go-version
```

Comment:
I'm not sure how useful it is to run all the checks and verifications every single time a make is run - just for the chance that someone without go will save some time.

How about a make check target - that users can run the first time ( maybe dependent on a .checked file ) ?

Reply:
This particular check that you've commented on is more about whether someone has a new enough go (vs whether they have one at all or not), and istio's requirement for go can change at any arbitrary time.  Stated differently it's as much about telling existing users they need to update to meet new requirements than it is about notifying new users they need to do more work to get things to work.

I think the way I implemented it is basically akin to a .checked file. In the common case "check-go-version"  should just check whether the appropriate "have_go_XXX" file exists in the bin/ directory.  If we start having more checks then it might get messy to continue the trend and end up with multiple special-purpose sentinel files (vs one generic "checked" file), though with a generic file it can be tricky to properly detect when the requirements have changed and a thus re-check should be performed.